scripts/testpack.sh

```diff
@@ -4,20 +4,26 @@
 ROOT="$(pwd)/$(dirname "$0")/.."
 cd "$ROOT" || exit 1
 
-npm pack
+TEST_DIR=$(mktemp -d)
 
-mkdir -p ../test-pack
+mkdir -p ${TEST_DIR}
 
-cp examples/node-commonjs/example.js ../test-pack/commonjs.js
-cp examples/node-esmodules/example.mjs ../test-pack/esmodules.mjs
-cp examples/node-esmodules/package.mjs ../test-pack/esmodules-package.mjs
+trap 'popd && rm -rf $TEST_DIR' EXIT
```

Comment:
I’d remove the popd from here… npm pack could fail and then you’d popd without having pushd’ed…

What we mostly care about with the trap is to no leave temp files.

If you want to overengineer, add a separate trap with just popd right after the pushd.